### ISSUE TITLE
[Bug]: sdk-react-native, Error resolving module `@metamask/sdk-react-native`

### ISSUE THREAD

**Issue by Thormeard** [MetaMask/metamask-sdk#929]
### SDK

React-Native

### Provide environment information

react-native: 0.74.2
iOS 17.3.1
@metamask/sdk-react-native: 0.3.1

### MetaMask SDK Version

"@metamask/sdk-react-native": "^0.3.1"

### MetaMask Mobile app Version

-

### Describe the Bug

Having this error with the latest `0.3.1` version of `sdk-react-native`. Rolling back to `0.3.0` fixes the issue.

Both android and ios. Reproducible with example https://github.com/MetaMask/metamask-sdk/tree/main/packages/examples/reactNativeSdkDemo

Similar issue in `sdk-react` from last year https://github.com/MetaMask/metamask-sdk/issues/596

```
[Error: InternalError Metro has encountered an error: While trying to resolve module `@metamask/sdk-react-native` from file `./src/components/MetaMask/MetaMaskConnector.js`, the package `./node_modules/@metamask/sdk-react-native/package.json` was successfully found. However, this package itself specifies a `main` module field that could not be resolved (`./node_modules/@metamask/sdk-react-native/index`. Indeed, none of these files exist:

  * ./node_modules/@metamask/sdk-react-native/index(.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx)
  * ./node_modules/@metamask/sdk-react-native/index/index(.ios.js|.native.js|.js|.ios.jsx|.native.jsx|.jsx|.ios.json|.native.json|.json|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx): ./node_modules/metro/src/node-haste/DependencyGraph.js (243:17)
```

### Expected Behavior

No issues when starting an example app

### To Reproduce

- Take example https://github.com/MetaMask/metamask-sdk/tree/main/packages/examples/reactNativeSdkDemo
- Update `@metamask/sdk-react-native` to `0.3.1`
- yarn android/ios

**Comment by omridan159** [comment]
@Thormeard 
Additionally, I have updated the example in our repo to version 0.3.1 in [this PR](https://github.com/MetaMask/metamask-sdk/pull/936). Please check it out and try to run it again after it gets merged.

**Comment by Thormeard** [comment]
@omridan159 

Thanks for the response.

I tried to do pod install with repo update flag, nothing changed.

Tried on another machine and with android build - same issue. Tried to start metro server with `--reset-cache` flag as well.

Also tried to delete `node_modules` and do yarn couple of time just to be sure.

**Comment by omridan159** [comment]
Hey @Thormeard, this issue should be resolved in the latest version: [v0.3.5](https://www.npmjs.com/package/@metamask/sdk-react-native/v/0.3.5).

Could you please give it another try and let us know how it goes? 🙏
